R/httr2-translate.R

```diff
@@ -0,0 +1,189 @@
+#' Translate httr2 request to curl command
+#'
+#' Convert an httr2 request object to equivalent curl command line syntax.
+#' This is useful for debugging, sharing requests, or converting to other tools.
+#'
+#' @param .req An httr2 request object created with [request()].
+#' @return A character string containing the curl command.
+#' @export
+#' @examples
+#' \dontrun{
+#' # Basic GET request
+#' request("https://httpbin.org/get") |>
+#'   httr2_translate()
+#'
+#' # POST with JSON body
+#' request("https://httpbin.org/post") |>
+#'   req_body_json(list(name = "value")) |>
+#'   httr2_translate()
+#'
+#' # POST with form data
+#' request("https://httpbin.org/post") |>
+#'   req_body_form(name = "value") |>
+#'   httr2_translate()
+#' }
+httr2_translate <- function(.req) {
+  # validate the request
+  check_request(.req)
+
+  # Extract URL
+  url <- .req$url
+
+  # use the request's method if it is set, otherwise infer
+  method <- .req$method %||%
+    {
+      if (!is.null(.req$body$data)) {
+        "POST"
+      } else {
+        "GET"
+      }
+    }
+
+  # we will append to cmd_parts to build up the request
+  cmd_parts <- c("curl")
+
+  # if the method isn't GET, it needs to be specified with `-X`
+  if (method != "GET") {
+    cmd_parts <- c(cmd_parts, paste0("-X ", method))
+  }
+
+  # if headers are present, add them using -H flag
+  if (!is.null(.req$headers) && length(.req$headers) > 0) {
+    headers <- .req$headers
+    for (name in names(headers)) {
+      value <- headers[[name]]
+
+      # handle weakrefs
+      if (rlang::is_weakref(value)) {
+        value <- rlang::wref_value(value)
+      }
+
+      # unobfuscate obfuscated
+      if (is_obfuscated(value)) {
+        value <- unobfuscate(value, handle = "reveal")
+      }
+
+      cmd_parts <- c(cmd_parts, paste0('-H "', name, ': ', value, '"'))
+    }
+  }
+
+  # manage options
+  if (!is.null(.req$options) && length(.req$options) > 0) {
+    options <- .req$options
+    for (name in names(options)) {
+      value <- options[[name]]
+      # convert options to curl flags
+      curl_flag <- switch(
+        name,
+        "timeout" = paste0("--max-time ", value),
+        "connecttimeout" = paste0("--connect-timeout ", value),
+        "proxy" = paste0("--proxy ", value),
+        "useragent" = paste0('--user-agent "', value, '"'),
+        "referer" = paste0('--referer "', value, '"'),
+        "followlocation" = if (value) "--location" else NULL,
+        "ssl_verifypeer" = if (!value) "--insecure" else NULL,
+        "verbose" = if (value) "--verbose" else NULL,
+        "cookiejar" = paste0('--cookie-jar "', value, '"'),
+        "cookiefile" = paste0('--cookie "', value, '"'),
+        # for unknown options try guess the flag if it was the intention
+        paste0("--", gsub("_", "-", name), " ", value)
+      )
+      if (!is.null(curl_flag)) {
+        cmd_parts <- c(cmd_parts, curl_flag)
+      }
```

Comment:
```suggestion
      cmd_parts <- c(cmd_parts, curl_flag)
```

`c(x, NULL)` returns `x`